src/gwt/src/org/rstudio/studio/client/workbench/views/terminal/TerminalSession.java

```diff
@@ -71,33 +74,38 @@ public TerminalSession(final ShellSecureInput secureInput,
       setHeight("100%");
       setTitle("Terminal " + sequence_);
    }
-   
+
    @Inject
    private void initialize(WorkbenchServerOperations server,
                            EventBus events)
    {
       server_ = server;
       eventBus_ = events; 
    } 
-   
+
    /**
     * Create a terminal process and connect to it.
     */
    public void connect()
    {
+      if (connected_ || connecting_)
+         return;
+
+      connecting_ = true;
       setNewTerminal(getHandle() == null);
-      
+
       server_.startTerminal(80, 25, getHandle(), getTitle(), getSequence(),
                             new ServerRequestCallback<ConsoleProcess>()
       {
          @Override
          public void onResponseReceived(ConsoleProcess consoleProcess)
          {
             consoleProcess_ = consoleProcess;
-            
+
```

Comment:
If `consoleProcess_` is null, it looks like we'll be left in the `connecting_` state indefinitely here. Intentional?